app/models/case_log.rb

```diff
@@ -215,7 +215,16 @@ def set_derived_fields
       self.month = startdate.month
       self.year = startdate.year
     end
-    self.incref = 1 if net_income_known == "Tenant prefers not to say"
+    case net_income_known
+    when "Yes – the household has a weekly income"
+      self.incfreq = "Weekly"
+    when "Yes – the household has a monthly income"
+      self.incfreq = "Monthly"
+    when "Yes – the household has a yearly income"
+      self.incfreq = "Yearly"
+    when "Tenant prefers not to say"
+      self.incref = 1
+    end
```

Comment:
I don't know if it'd look better or not, but what if we have a mapping instead?
```INCREF_MAPPINGS =  {"Yes – the household has a weekly income" : "Weekly", 
"Yes – the household has a monthly income": "Monthly" ...}

self.incfrec = INCREF_MAPPINGS[net_income_known]
```

Reply:
Happy to change but note there's two different fields being set there depending on the case (`incref` and incfreq` so might not get rid of that much)

Reply:
Oh, yeah, that's fair. I don't have a strong preference